### DISCUSSION TITLE
Needs Help setting up distcc & ssh

### DISCUSSION THREAD

**Discussion by ghost** [distcc/distcc#526] [Q&A] [answered]
sorry for garbage post, here is cleanup.

daemon config:
Man page: "For SSH connections, distccd must be installed on the volunteer but should not run as a daemon -- it will be started over SSH as needed." So should i disregard any daemon config files from distccd?

SSH to what User:
https://wiki.gentoo.org/wiki/Distcc#SSH_for_communication Is unclear which user should be used is any user fine or should distcc user be used.

Cross compilation can't find cross compiler
`distcc[6195] (dcc_execvp) ERROR: failed to exec i686-pc-linux-musl-gcc: No such file or directory)` 
https://wiki.gentoo.org/wiki/Distcc/Cross-Compiling tries to address it but ngl i haven't gotten it to work.

**Comment by sourcefrog** [comment, marked as answer]
From memory the daemon config is not read when you use ssh.

You can use any user that exists on the server.

The question about finding the cross compiler is too vague. Does that binary exist on the server? If you SSH to that user, is it on the path?

**Reply by ghost** [reply]
The cross compiler exists and is in $PATH (added/exported through the user's bashrc) but it does not seem to be found whenever distcc tries to find it. Does it not source the bashrc?

**Reply by sourcefrog** [reply]
bash has complex rules for which startup files are read: see https://www.gnu.org/software/bash/manual/bash.html#Bash-Startup-Files

In the `distcc - ssh - sshd - bash - distccd` path, the bash is probably not an "interactive shell" and so `.bashrc` won't be read.

Check that you can run the compiler with just `ssh someuser@whateverhost i686-pc-linux-musl-gcc`. If that doesn't work then distcc can't fix it. Maybe fix the path in some other file that bash does load for interactive hosts.

**Reply by ghost** [reply]
I can `ssh srv` then launch `i686-pc-linux-musl-gcc`
but i can not do `ssh srv i686-pc-linux-musl-gcc`
I guess i have to read the ssh documentation now 

The non interactive shell does not source my bashrc

**Reply by sourcefrog** [reply]
Yep, that's the interactive vs noninteractive shell. It's unfortunately complicated in bash, but if you get `ssh srv i686-pc-linux-musl-gcc` working then distcc over ssh should also work.

**Reply by ghost** [reply]
Thanks for all the help
btw 
`ssh srv "echo $PATH"` is not the same as `ssh srv 'echo $PATH'` - this ruined my day
